sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -126,25 +148,114 @@
 
   @Override
   public Span startSpan() {
-    // TODO: get remoteParent span from the context if noParent=false and remoteParent/remoteParents
-    // are null
-    // TODO: correctly implement this.
-    SpanContext context =
-        SpanContext.create(
-            TraceId.fromBytes(new byte[] {1, 2, 3, 4, 5, 6, 7, 8, 8, 7, 6, 5, 4, 3, 2, 1}, 0),
-            SpanId.fromBytes(new byte[] {1, 2, 3, 4, 5, 6, 7, 8}, 0),
-            TraceOptions.builder().setIsSampled(true).build(),
-            Tracestate.getDefault());
+    Span currentSpan = null;
```

Comment:
currentSpan logic can be moved in the parent(...) method. This simplifies the code and avoid an unnecessary if jump.